docs/uenv-ddt.md

```diff
@@ -0,0 +1,124 @@
+
+## Using the debugger
+
+In order to debug your code on Alps, you need to:
+
+- setup the user environment on Alps,
+- install the DDT client on your laptop,
+- build an executable with debug flags on Alps,
+- launch a job with the debugger on Alps.
+
+### Setup the environment
+
+`gcc113-mpich8125-cuda118-forge2301.squashfs` provides the latest version of Linaro Forge (23.0.1), along with gcc/11.3, cray-mpich-8.1.25 and cuda/11.8.
+
+- On Alps:
+```bash
+uenv start ./gcc113-mpich8125-cuda118-forge2301.squashfs
+ls -l /user-environment/
+source /user-environment/env/default/activate.sh
+```
+
+### Install the client
+
+We recommend to download and install the desktop [client](https://www.linaroforge.com/downloadForge) on your local workstation. It will connect with the debug jobs running on Alps, offering a better user experience compared to opening ddt with X11 forwarding. The client can be downloaded for a selection of operating systems.
+
+Once installed, the client needs to be configured to connect to your preferred vcluster. For this, launch the client:
+
+```
+- mac: open /Applications/Linaro\ Forge\ Client\ 23.0.1.app/
+- linux: $HOME/linaro/forge/23.0.1/bin/ddt
+```
+
+and setup the connection:
+
+```
+- open the 'Remote Launch' menu and click on 'configure' and then 'Add',
+- set the fields, for example:
+    - Connection Name: alps
+    - Host Name: piccinal@ela.cscs.ch piccinal@hohgant.cscs.ch
+      # replace piccinal with your CSCS username
+    - Remote install dir: /usr/bin/squashfs-run IMG DDTDIR
+      # here we tell the client to use the ddt installed in the uenv image
+    - Remote Script:
+    - Private Key: /Users/piccinal/.ssh/cscs-key
+      # replace /Users/piccinal with your laptop username
+    - Proxy through login node: yes (check the box)
+```
+
+where you will replace IMG and DDTDIR with for example:
+```
+- IMG is /scratch/e1000/piccinal/hohgant/octree/gcc113-mpich8125-forge2301.squashfs
+- DDTDIR is /user-environment/linux-sles15-zen2/gcc-11.3.0/ddt-23.0.1-qobm4ghhupq4ueij3kkdtzozd5qevjpl
+```
+
+Click `Test Remote Launch`. If the client can connect, you are ready to debug:
+click on `ok` and `close` (to save the configuration). You can now connect by going to `Remote Launch` and choose the `Alps` entry. If the client fails to connect, look at the message, check your ssh configuration and make sure you can ssh without the client.
+
+### Build with debug flags
+
+Once the uenv is loaded and activated, the program to debug must be compiled with the `-g` (for cpu) and `-G` (for gpu) debugging flags. For example, download the example code:
+
+- on Alps:
+```bash
+git clone https://github.com/sekelle/octree-miniapp octree-miniapp.git
+wget https://raw.githubusercontent.com/jgphpc/octree-miniapp/ddt/neighbor_search.patch
+cd octree-miniapp.git
+patch -i neighbor_search.patch
+cd ..
+```
+
+build the code with:
+ 
+- on Alps:
```

Comment:
This example of "how to build" is quite complicated and specific to an example of building a cuda application. It includes a "trick" to collect the headers and pass them to nvcc and how to use the cuda stubs.

In the real world, this type of configuration would be handled (hopefully!) by CMake and friends. I'm worried that writing these docs like this might confuse users into thinking that this is how _all_ code should be compiled.

We could make additional documentation for tricks like these ones, that shows how and why each method is employed?